### PR TITLE
Feature/props in formstore state

### PR DESCRIPTION
**Pull request by MartynasZilinskas**

### CODE REVIEW

File: packages/simplr-forms-core/src/contracts/field.ts [hunk 1]
```diff
@@ -30,9 +30,13 @@ export interface FieldState {
     FieldsGroup?: {
         Id: string;
     };
+    Props?: FieldStatePropsRecord;
 }
 
+export type FieldStateProps = FieldProps & React.Props<void>;
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms-core/src/contracts/field.ts]: Change to any

File: packages/simplr-forms-core/__tests__/stores/form-store.test.ts [hunk 2]
```diff
@@ -150,4 +156,40 @@ describe("Form store", () => {
             done.fail(error);
         }
     });
+
+    it("registers field with props", () => {
+        const formId = "FORM-ID";
+        const fieldId = "FIELD-ID";
+        const fieldProps: MyFieldProps = {
+            name: "fieldName",
+            value: "initial-value"
```
Comment (DovydasNavickas) [inline packages/simplr-forms-core/__tests__/stores/form-store.test.ts]: Add `randomKey` with `random value` for a non-default key: value situation check.

File: packages/simplr-forms-core/__tests__/stores/form-store.test.ts [hunk 3]
```diff
@@ -150,4 +156,40 @@ describe("Form store", () => {
...
+        const fieldProps: MyFieldProps = {
+            name: "fieldName",
+            value: "initial-value"
+        };
+        const formStore = new FormStore(formId);
+
+        formStore.RegisterField(fieldId, fieldProps.value, fieldProps);
+
+        const fieldPropsRecord = recordify<FieldStateProps, FieldStatePropsRecord>(fieldProps);
+        // Check deeply if it's the equal object.
```
Comment (DovydasNavickas) [inline packages/simplr-forms-core/__tests__/stores/form-store.test.ts]: Check deeply -> Deep-check the updated props

File: packages/simplr-forms-core/__tests__/stores/form-store.test.ts [hunk 4]
```diff
@@ -150,4 +156,40 @@ describe("Form store", () => {
...
+
+    it("updates field props", () => {
+        const formId = "FORM-ID";
+        const fieldId = "FIELD-ID";
+        const fieldProps: MyFieldProps = {
+            name: "fieldName",
+            value: "initialValue"
+        };
+        const fieldPropsNext: MyFieldProps = {
+            name: "fieldName",
```
Comment (DovydasNavickas) [inline packages/simplr-forms-core/__tests__/stores/form-store.test.ts]: `name: fieldProps.name`

File: packages/simplr-forms-core/src/contracts/field.ts [hunk 5]
```diff
@@ -17,7 +17,7 @@ export interface FieldProps {
     validationType?: FieldValidationType;
     onBlur?: (event: any) => void;
     onFocus?: (event: any) => void;
-    children?: React.ReactNode;
+    // children?: React.ReactNode;
```
Comment (DovydasNavickas) [inline packages/simplr-forms-core/src/contracts/field.ts]: Remove comment.

File: packages/simplr-forms-core/src/stores/form-store.ts [hunk 6]
```diff
@@ -98,6 +106,21 @@ export class FormStore extends ActionEmitter {
         return this.State.Fields.get(fieldId);
     }
 
+    public UpdateProps(fieldId: string, props: FieldProps) {
+        const propsRecord = recordify<FieldStateProps, FieldStatePropsRecord>(props);
+        const fieldState = this.State.Fields.get(fieldId);
+
+        if (fieldState.Props != null &&
```
Comment (DovydasNavickas) [inline packages/simplr-forms-core/src/stores/form-store.ts]: Down-level if.

**Pull request merged**